src/main/java/roomescape/dao/ReservationDao.java

```diff
@@ -1,10 +1,145 @@
 package roomescape.dao;
 
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.time.LocalDate;
 import java.util.List;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
 import roomescape.model.Reservation;
+import roomescape.model.ReservationTime;
+import roomescape.model.Theme;
 
-public interface ReservationDao {
-    public Reservation save(Reservation reservation);
-    public boolean deleteById(Long id);
-    public List<Reservation> findAll();
+@Repository
+public class ReservationDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public ReservationDao(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private static RowMapper<Reservation> reservationRowMapper() {
+        return (rs, rowNum) -> new Reservation(
+                rs.getLong("reservation_id"),
+                rs.getString("name"),
+                rs.getDate("date").toLocalDate(),
+                new ReservationTime(
+                        rs.getLong("time_id"),
+                        rs.getTime("time_value").toLocalTime()
+                ),
+                new Theme(
+                        rs.getLong("theme_id"),
+                        rs.getString("name"),
+                        rs.getString("description"),
+                        rs.getString("thumbnail")
+                )
```

Comment:
```suggestion
        return (rs, rowNum) -> {
            var reservationTime = new ReservationTime(
                    rs.getLong("time_id"),
                    rs.getTime("time_value").toLocalTime()
            );

            var theme = new Theme(
                    rs.getLong("theme_id"),
                    rs.getString("name"),
                    rs.getString("description"),
                    rs.getString("thumbnail")
            );

            return new Reservation(
                    rs.getLong("reservation_id"),
                    rs.getString("name"),
                    rs.getDate("date").toLocalDate(),
                    reservationTime,
                    theme
            );
        };
```

위와 같이 변수 선언을 분리해보면 어떨까요?